linodecli/plugins/get-kubeconfig.py

```diff
@@ -0,0 +1,168 @@
+    )
+    currentConfig = None
+
+    if os.path.exists(os.path.expanduser(parsed.kubeconfig)):
+        currentConfig = _load_config(os.path.expanduser(parsed.kubeconfig))
+
+    # If there is no current kubeconfig, dump the cluster config to the specified file location.
+    # If there is a current kubeconfig, merge it with the cluster's kubeconfig
+    clusterConfig = (
+        _merge_dict(currentConfig, clusterConfig)
+        if currentConfig is not None
+        else clusterConfig
+    )
+    if parsed.dry_run:
+        print(clusterConfig)
+    else:
+        _dump_config(os.path.expanduser(parsed.kubeconfig), clusterConfig)
+
+
+# Fetches the kubeconfig of the lke cluster with the specified label
+def _get_kubeconfig_by_label(cluster_label, client):
+    """
+    Returns the LKE Cluster with the given Label
+    """
+
+    code, cluster = client.call_operation(
+        "lke", "clusters-list", args=["--label", cluster_label]
+    )
+
+    if code != 200:
+        print(f"Error retrieving cluster: {code}", file=sys.stderr)
+        sys.exit(1)
+
+    if len(cluster["data"]) == 0:
+        print(
+            f"Cluster with label {cluster_label} does not exist.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    code, kubeconfig = client.call_operation(
+        "lke", "kubeconfig-view", args=[str(cluster["data"][0]["id"])]
+    )
+
+    if code != 200:
+        print(f"Error retrieving kubeconfig: {code}", file=sys.stderr)
+        sys.exit(1)
+
+    return kubeconfig
+
+
+# Loads a yaml file
+def _load_config(filepath):
+    with open(filepath, "r", encoding="utf-8") as file_descriptor:
+        data = yaml.load(file_descriptor, Loader=yaml.Loader)
+
+    if not data:
+        print(f"Could not load file at {filepath}", file=sys.stderr)
+        sys.exit(1)
+
+    return data
+
+
+# Dumps data to a yaml file
+def _dump_config(filepath, data):
+    os.makedirs(os.path.dirname(filepath), exist_ok=True)
+    with open(filepath, "w", encoding="utf-8") as file_descriptor:
+        yaml.dump(data, file_descriptor)
+
+
+# Merges the lists in the provided dicts. If non-list properties of the two
+# dicts differ, uses the value from the first dict.
+def _merge_dict(dict_1, dict_2):
+    for key in dict_1:
+        if not isinstance(dict_1[key], list):
+            continue
+
+        # Extract names of each list item for both dicts
+        dict1_names = [sub["name"] for sub in dict_1[key]]
+        dict2_names = [sub["name"] for sub in dict_2[key]]
+
+        new_entries = []
+
+        # Add all dict_2 entries with unique names to new_entries
+        for index, value in enumerate(dict2_names):
+            if value not in dict1_names:
+                new_entries.append(dict_2[key][index])
+
+        # Concat dict_1 list with new_entries and update dict_1 list to concatenated list
+        merged = dict_1[key] + new_entries
+        dict_1[key] = merged
+
+    return dict_1
```

Comment:
We can optimize this a bit using a map to merge values by name:
```python
def _merge_dict(dict_1, dict_2):
    # Return a new dict to prevent any accidental mutations
    result = {}

    for key in dict_1:
        if not isinstance(dict_1[key], list):
            result[key] = dict_1[key]
            continue

        merge_map = {
            sub["name"]: sub for sub in dict_1[key]
        }

        for sub in dict_2[key]:
            # If the name is already in the merge map, skip
            if sub["name"] in merge_map:
                continue

            merge_map[sub["name"]] = sub

        # Convert back to a list
        result[key] = [v for v in merge_map.values()]

    return result
```

Reply:
Nice! Thanks for the tip